Assets/Examples/Scripts/DemoUX.cs

```diff
@@ -0,0 +1,364 @@
+                }
+            }
+
+            if (selectedTile.Parent != null && Input.GetKeyDown(KeyCode.UpArrow))
+            {
+                selectedStack.Push(selectedTile);
+                selectedTile = selectedTile.Parent;
+            }
+
+            if (selectedTile.Children.Count > 0 && Input.GetKeyDown(KeyCode.DownArrow))
+            {
+                selectedTile = selectedStack.Count > 0 ? selectedStack.Pop() : selectedTile.Children.First();
+            }
+
+            if (selectedTile.Parent != null && Input.GetKeyDown(KeyCode.LeftArrow))
+            {
+                int idx = selectedTile.Parent.Children.FindIndex(c => c == selectedTile) - 1;
+                if (idx < 0)
+                {
+                    idx = selectedTile.Parent.Children.Count -1;
+                }
+                selectedTile = selectedTile.Parent.Children[idx];
+            }
+
+            if (selectedTile.Parent != null && Input.GetKeyDown(KeyCode.RightArrow))
+            {
+                int idx = selectedTile.Parent.Children.FindIndex(c => c == selectedTile) + 1;
+                if (idx == selectedTile.Parent.Children.Count)
+                {
+                    idx = 0;
+                }
+                selectedTile = selectedTile.Parent.Children[idx];
+            }
+
+            selectedTile.Tileset.Traversal.ForceTiles.Clear();
+            selectedTile.Tileset.Traversal.ForceTiles.Add(selectedTile);
+        }
+
+        //toggle hud
+        if (hud != null && Input.GetKeyDown(KeyCode.H))
+        {
+            hud.text.enabled = !hud.text.enabled;
+        }
+
+        if (Input.GetKeyDown(KeyCode.V))
+        {
+            ResetView();
+        }
+
+        if (Input.GetKeyDown(KeyCode.F))
+        {
+            FitView();
+        }
+
+        //toggle nav mode
+        if (mouseFly != null && mouseRotate != null && Input.GetKeyDown(KeyCode.N))
+        {
+            mouseFly.enabled = !mouseFly.enabled;
+            mouseRotate.enabled = !mouseRotate.enabled;
+        }
+
+        //set rotNav pivot
+        if (rotNav && Input.GetKeyDown(KeyCode.C))
+        {
+            if (hasPick)
+            {
+                mouseRotate.pivot = pointer.transform.position;
+            }
+            else if (tileset && tileset.Ready())
+            {
+                mouseRotate.pivot = tileset.transform.TransformPoint(tileset.BoundingSphere().position);
+            } 
+        }
+
+        //handle mouse clicks
+        if (hasFocus && !MouseNavBase.MouseOnUI())
+        {
+            if (Input.GetMouseButtonDown(0))
+            {
+                lastMouse = Input.mousePosition;
+            }
+            else if (!Input.GetMouseButton(0) && lastMouse != null && lastMouse.Value == Input.mousePosition)
+            {
+                OnClick(Input.mousePosition);
+                lastMouse = null;
+            }
+        }
+        else
+        {
+            lastMouse = null;
+        }
```

Comment:
Will this register a click if I drag and then pause before releasing mouse? I'm thinking would need to set "firstMouse" on mouse down only if lastMouse is null. Then check mousePosition against firstMouse for click

Reply:
I think there was a bug here, idk if this is exactly what you were talking about, but it would have registered a click if the user pressed the mouse down, then moved it around, then returned to the same pixel they pressed on, then let the mouse up.  I pushed a fix for that.